clients/src/test/java/org/apache/kafka/common/metrics/JmxReporterTest.java

```diff
@@ -76,6 +77,21 @@ public void testJmxRegistration() throws Exception {
         }
     }
 
+    @Test
+    public void testMbeanTagOrdering() {
+        Map<String, String> tags = new HashMap<>();
+        tags.put("tag_a", "x");
+        tags.put("tag_b", "y");
+        tags.put("tag_c", "z");
+        tags.put("tag_d", "1,2");
+        tags.put("tag_e", "");
+        tags.put("tag_f", "3");
```

Comment:
Although `HashMap` doesn't guarantee the element order, the current test might make people think this test is passed due to the element insertion is key-ordered. Could you put the key in un-orderded sequence, so that we can make sure your fix works?

Reply:
the test already fails if I remove the ordering code, but I can change the order if that makes us feel better.